deps/v8/src/base/platform/platform-posix.cc

```diff
@@ -321,15 +321,11 @@ int OS::GetCurrentProcessId() {
 
 
 int OS::GetCurrentThreadId() {
-#if V8_OS_MACOSX
-  return static_cast<int>(pthread_mach_thread_np(pthread_self()));
-#elif V8_OS_LINUX
+#if defined(ANDROID)
   return static_cast<int>(syscall(__NR_gettid));
-#elif V8_OS_ANDROID
-  return static_cast<int>(gettid());
 #else
-  return static_cast<int>(pthread_self());
-#endif
+  return static_cast<int>(syscall(SYS_gettid));
+#endif  // defined(ANDROID)
```

Comment:
We probably want to float 3589a62104eac1daae782ab479bec09f4df4dc9a again, or cherry-pick another fix from upstream like https://github.com/v8/v8-git-mirror/commit/68d5f916d885462621d52ceafdd605ce1fd28c24.